Kentor.AuthServices/Configuration/Options.cs

```diff
@@ -24,15 +24,23 @@ public static Options FromConfiguration
         {
             get
             {
-                var options = new Options(KentorAuthServicesSection.Current);
-                KentorAuthServicesSection.Current.IdentityProviders.RegisterIdentityProviders(options);
-                KentorAuthServicesSection.Current.Federations.RegisterFederations(options);
-                options.SPOptions.ServiceCertificate = KentorAuthServicesSection.Current.ServiceCertificateConfiguration.LoadCertificate();
-
-                return options;
+                return optionsFromConfiguration.Value;
             }
         }
 
+        private static readonly Lazy<Options> optionsFromConfiguration 
+            = new Lazy<Options>(() => LoadOptionsFromConfiguration(), false);
```

Comment:
I think it should be threadsafe (passing true instead). The config might be accessed by several threads on a busy server.